contracts/aave/AaveV2Contract.sol

```diff
@@ -0,0 +1,173 @@
+        depositAssets[_asset] = _priceOracle;
+        emit AddAsset(_asset,_priceOracle);
+    }
+
+    /**
+     * @dev Users will deposit their assets in pool
+     * @param _asset will be asset name like ETH or BTC
+     * @param _amount will be the deposited amount
+     */
+    function deposit( address _asset,uint256 _amount) public nonReentrant payable {
+        require(
+            depositAssets[_asset] != address(0),
+            Errors.ASSET_NOT_ALLOWED
+        );
+
+        int256 assetPrice = getLatestPrice(depositAssets[_asset]);
+
+        if (msg.value > 0) {
+            WETHGATEWAY.depositETH{value: msg.value}(
+                address(POOL),
+                address(this),
+                0
+            );
+        } else {
+            require(_amount > 0, Errors.ZERO_AMOUNT);
+            IERC20 supplyAsset = IERC20(_asset);
+            supplyAsset.transferFrom(msg.sender, address(this), _amount);
+            supplyAsset.approve(address(POOL), _amount);
+            POOL.deposit(address(supplyAsset), _amount, address(this), 0);
+        }
+    }
+
+    /**
+     * @dev Cruize will borrow on behalf of user
+     * @param _asset will be asset name like ETH or BTC
+     */
+    function borrow(address _asset) public nonReentrant {
+        (, , uint256 availableBorrowsETH, , , ) = POOL.getUserAccountData(
+            address(this)
+        );
+
+        int256 assetPrice = getLatestPrice(depositAssets[_asset]);
+        uint256 availableBorrowAmountIn18Decimals = (nomalize(
+            uint256(assetPrice)
+        ) * availableBorrowsETH) / 1e18;
+        uint256 availableBorrowAmountIn6Decimals = availableBorrowAmountIn18Decimals /
+                1e12;
+        uint256 borrowAmount = (availableBorrowAmountIn6Decimals *
+            borrowRatio) / 1000;
+        POOL.borrow(address(BORROW_ASSET), borrowAmount, 2, 0, address(this));
+    }
+
+    /**
+     * @dev Cruize pool will repay the debt amount
+     */
+    function repay() public nonReentrant {
+        BORROW_ASSET.approve(address(POOL), type(uint256).max);
+        POOL.repay(address(BORROW_ASSET), type(uint256).max, 2, address(this));
+    }
+
+    /**
+     * @dev Criuze will withdraw collateral amount
+     * @param _asset will be asset name like ETH or BTC
+     */
+    function withdraw(address _asset) public nonReentrant {
+        (, uint256 debt, , , , ) = POOL.getUserAccountData(address(this));
+        POOL.withdraw(
+            _asset,
+            type(uint256).max,
+            address(this)
+        );
+    }
+
+    /**
+     * @dev only owner can change the borrow ratio
+     * @param ratio percentage in 1000 bips i.e 100 == 10% of 1000
+     * @return event BorrowRatioChanged(ratio = 200)
+     */
+    function changeBorrowRatio(uint256 ratio) public onlyOwner {
+        require(ratio != borrowRatio, Errors.BORROW_NOT_CHANGED);
+        borrowRatio = ratio;
+        emit BorrowRatioChanged(ratio);
+    }
+}
```

Comment:
also, create a fallback function and don't allow users to directly  sent ETH to contract or any asset to contract